packages/x-date-pickers/src/internals/hooks/useField/useField.interfaces.ts

```diff
@@ -82,13 +83,11 @@ export interface UseFieldForwardedProps {
   onClick?: () => void;
   onFocus?: () => void;
   onBlur?: () => void;
+  onPaste?: React.ClipboardEventHandler;
+  inputProps?: Partial<InputBaseProps['inputProps']>;
 }
 
-export type UseFieldResponse<TForwardedProps extends UseFieldForwardedProps> = Omit<
-  TForwardedProps,
-  keyof UseFieldForwardedProps
-> &
-  Required<UseFieldForwardedProps> &
+export type UseFieldResponse = Required<UseFieldForwardedProps> &
```

Comment:
This is the most interesting change.
Could you take a look at it @flaviendelangle ?
I've tried to untangle logic in here, but I couldn't understand  the reason why the previous type def was needed. It started erroring out for me when I introduced `inputProps` in `UseFieldForwardedProps `

Reply:
[This](https://github.com/mui/mui-x/pull/7135/files#diff-968c508c9db1c5aa01a8484bd478e639c7ea15f455614de20f612a50cf7b8a84R91) seems to have solved the typing issue. The reason why it did help is still unclear to me when computing the resulting types in my brain 🙈 🤷